## Modis folder analysis



In [1]:
import numpy as np
import pandas as pd
import os
import matplotlib.pyplot as plt
from glob import glob
from tqdm import tqdm
from PM25_data_loader_analysis import Modis_data_loader
from PM25_data_loader_analysis import PM_25_dataloader
from PM25_data_loader_analysis import combine_the_data_frames
from PM25_data_loader_analysis import Training_data_loader

### Modis data analysis

In [2]:
Modis_data_2018 = Modis_data_loader("MODIS_AOD/merged_data_2018_sorted_dates.csv")

In [3]:
Modis_data_2018.get_head()
# The data is of 365 days latitude and longitude hence the )

latitude  longitude  2018-01-01  2018-01-02  2018-01-03  2018-01-04  \
0     28.414     77.171         NaN       0.322         NaN       0.816   
1     28.414     77.181         NaN       0.305         NaN         NaN   
2     28.414     77.192         NaN       0.293         NaN         NaN   
3     28.414     77.202         NaN       0.305         NaN         NaN   
4     28.414     77.213         NaN       0.319         NaN         NaN   
5     28.414     77.223         NaN       0.331         NaN         NaN   
6     28.424     77.171         NaN       0.328         NaN       0.806   
7     28.424     77.181         NaN       0.328         NaN       0.807   
8     28.424     77.192         NaN       0.317         NaN         NaN   
9     28.424     77.202         NaN       0.307         NaN         NaN   
10    28.424     77.213         NaN       0.341         NaN         NaN   
11    28.424     77.223         NaN       0.348         NaN         NaN   
12    28.424     77.233         NaN       0.327         NaN         NaN   
13    28.424     77.244         NaN       0.310         NaN       0.910   
14    28.434     77.161         NaN       0.393         NaN       0.774   
15    28.434     77.171         NaN       0.356         NaN       0.793   
16    28.434     77.181         NaN       0.327         NaN       0.806   
17    28.434     77.192         NaN       0.311         NaN       0.803   
18    28.434     77.202         NaN       0.300         NaN       0.811   
19    28.434     77.213         NaN       0.314         NaN         NaN   

    2018-01-05  2018-01-06  2018-01-07  2018-01-08  ...  2018-12-22  \
0        1.422       2.179       1.117         NaN  ...       0.552   
1        1.360       2.154       1.069         NaN  ...       0.543   
2        1.337       2.130       1.059         NaN  ...       0.550   
3        1.314       2.129       1.047         NaN  ...       0.570   
4        1.327       2.147       1.050         NaN  ...       0.591   
5        1.362       2.163       1.085         NaN  ...       0.610   
6        1.476       2.159       1.157         NaN  ...       0.546   
7        1.430       2.141       1.116         NaN  ...       0.554   
8        1.372       2.134       1.079         NaN  ...       0.550   
9        1.348       2.115       1.070         NaN  ...       0.549   
10       1.352       2.136       1.092         NaN  ...       0.595   
11       1.376       2.148       1.117         NaN  ...         NaN   
12       1.363       2.138       1.094         NaN  ...       0.612   
13       1.327       2.113       1.078         NaN  ...       0.608   
14       1.614       2.298       1.197         NaN  ...       0.520   
15       1.582       2.208       1.198         NaN  ...       0.533   
16       1.533       2.121       1.157         NaN  ...       0.546   
17       1.434       2.065       1.096         NaN  ...       0.554   
18       1.381       2.060       1.092         NaN  ...       0.552   
19       1.353       2.082       1.115         NaN  ...       0.573   

    2018-12-23  2018-12-24  2018-12-25  2018-12-26  2018-12-27  2018-12-28  \
0        0.933       1.650       0.921         NaN       0.554         NaN   
1        0.936       1.620       0.884         NaN       0.522         NaN   
2        0.954       1.626       0.870         NaN       0.516         NaN   
3        0.973       1.654       0.863         NaN       0.533         NaN   
4        1.009       1.703       0.879         NaN       0.579         NaN   
5        1.056       1.716       0.910         NaN       0.616         NaN   
6        0.980       1.696       1.004         NaN       0.597         NaN   
7        0.961       1.666       0.937         NaN       0.569         NaN   
8        0.951       1.634       0.887         NaN       0.535         NaN   
9        0.965       1.622       0.885         NaN       0.527         NaN   
10       1.036       1.701       0.915         NaN       0.601         NaN   


In [4]:
Modis_data_2018.get_shape()

(1350, 367)

In [5]:
Modis_data_2019 = Modis_data_loader("MODIS_AOD/merged_data_2019_sorted_dates.csv")
Modis_data_2019.get_head()


latitude  longitude  2019-01-01  2019-01-02  2019-01-03  2019-01-04  \
0     28.414     77.171         NaN         NaN         NaN         NaN   
1     28.414     77.181         NaN         NaN         NaN         NaN   
2     28.414     77.192         NaN         NaN         NaN         NaN   
3     28.414     77.202         NaN         NaN         NaN         NaN   
4     28.414     77.213         NaN         NaN         NaN         NaN   
5     28.414     77.223       1.149         NaN         NaN         NaN   
6     28.424     77.171         NaN         NaN         NaN         NaN   
7     28.424     77.181         NaN         NaN         NaN         NaN   
8     28.424     77.192         NaN         NaN         NaN         NaN   
9     28.424     77.202         NaN         NaN         NaN         NaN   
10    28.424     77.213         NaN         NaN         NaN         NaN   
11    28.424     77.223         NaN         NaN         NaN         NaN   
12    28.424     77.233       1.153         NaN         NaN         NaN   
13    28.424     77.244       1.166         NaN         NaN         NaN   
14    28.434     77.161         NaN         NaN         NaN         NaN   
15    28.434     77.171         NaN         NaN         NaN         NaN   
16    28.434     77.181         NaN         NaN         NaN         NaN   
17    28.434     77.192       1.131         NaN         NaN         NaN   
18    28.434     77.202       1.135         NaN         NaN         NaN   
19    28.434     77.213       1.153         NaN         NaN         NaN   

    2019-01-05  2019-01-06  2019-01-07  2019-01-08  ...  2019-12-22  \
0          NaN         NaN         NaN       1.132  ...         NaN   
1        1.930         NaN         NaN       1.105  ...         NaN   
2        1.910         NaN         NaN       1.101  ...         NaN   
3        1.925         NaN       0.709       1.104  ...         NaN   
4        1.959         NaN         NaN       1.135  ...         NaN   
5        1.987         NaN         NaN       1.174  ...         NaN   
6          NaN         NaN         NaN       1.182  ...         NaN   
7        2.004         NaN         NaN       1.136  ...         NaN   
8        1.958         NaN         NaN       1.106  ...         NaN   
9        1.938         NaN         NaN       1.104  ...         NaN   
10       1.992         NaN         NaN       1.159  ...         NaN   
11       2.006         NaN         NaN       1.186  ...         NaN   
12       2.008         NaN         NaN       1.205  ...         NaN   
13       1.995         NaN       0.743       1.185  ...         NaN   
14         NaN         NaN         NaN       1.250  ...         NaN   
15         NaN         NaN         NaN       1.246  ...         NaN   
16       2.063         NaN         NaN       1.222  ...         NaN   
17       1.987         NaN         NaN       1.146  ...         NaN   
18       1.961         NaN         NaN       1.136  ...         NaN   
19       1.967         NaN         NaN       1.143  ...         NaN   

    2019-12-23  2019-12-24  2019-12-25  2019-12-26  2019-12-27  2019-12-28  \
0          NaN         NaN         NaN       0.531         NaN         NaN   
1          NaN         NaN         NaN       0.658         NaN         NaN   
2          NaN         NaN         NaN       0.636         NaN         NaN   
3          NaN         NaN         NaN       0.608         NaN         NaN   
4          NaN         NaN         NaN       0.615         NaN         NaN   
5          NaN         NaN         NaN       0.633         NaN         NaN   
6          NaN         NaN         NaN       0.634         NaN         NaN   
7          NaN         NaN         NaN       0.609         NaN         NaN   
8          NaN         NaN         NaN       0.580         NaN         NaN   
9          NaN         NaN         NaN       0.629         NaN         NaN   
10         NaN         NaN         NaN       0.692         NaN         NaN   


In [6]:
Modis_data_2019.get_shape()

(1350, 367)

In [7]:
Modis_data_2020 = Modis_data_loader("MODIS_AOD/merged_data_2020_sorted_dates.csv")
Modis_data_2020.get_head()

latitude  longitude  2020-01-01  2020-01-02  2020-01-03  2020-01-04  \
0     28.414     77.171         NaN         NaN       1.428       1.092   
1     28.414     77.181         NaN         NaN       1.389       1.077   
2     28.414     77.192         NaN         NaN       1.325       1.085   
3     28.414     77.202         NaN         NaN       1.306       1.106   
4     28.414     77.213         NaN         NaN       1.312       1.157   
5     28.414     77.223         NaN         NaN       1.301         NaN   
6     28.424     77.171         NaN         NaN       1.462       1.137   
7     28.424     77.181         NaN         NaN       1.448       1.114   
8     28.424     77.192         NaN         NaN       1.423       1.101   
9     28.424     77.202         NaN         NaN       1.378       1.099   
10    28.424     77.213         NaN         NaN       1.356       1.181   
11    28.424     77.223         NaN         NaN       1.343       1.239   
12    28.424     77.233         NaN         NaN       1.308       1.275   
13    28.424     77.244         NaN         NaN       1.263       1.245   
14    28.434     77.161         NaN         NaN       1.459       1.191   
15    28.434     77.171         NaN         NaN       1.471       1.186   
16    28.434     77.181         NaN         NaN       1.466       1.172   
17    28.434     77.192         NaN         NaN       1.431       1.112   
18    28.434     77.202         NaN         NaN       1.420       1.096   
19    28.434     77.213         NaN         NaN       1.407       1.103   

    2020-01-05  2020-01-06  2020-01-07  2020-01-08  ...  2020-12-22  \
0          NaN         NaN         NaN         NaN  ...         NaN   
1          NaN         NaN         NaN         NaN  ...         NaN   
2          NaN         NaN         NaN         NaN  ...         NaN   
3          NaN         NaN         NaN         NaN  ...         NaN   
4          NaN         NaN         NaN         NaN  ...         NaN   
5          NaN         NaN         NaN         NaN  ...         NaN   
6          NaN         NaN         NaN         NaN  ...         NaN   
7          NaN         NaN         NaN         NaN  ...         NaN   
8          NaN         NaN         NaN         NaN  ...         NaN   
9          NaN         NaN         NaN         NaN  ...         NaN   
10         NaN         NaN         NaN         NaN  ...         NaN   
11         NaN         NaN         NaN         NaN  ...         NaN   
12         NaN         NaN         NaN         NaN  ...         NaN   
13         NaN         NaN         NaN         NaN  ...         NaN   
14         NaN         NaN         NaN         NaN  ...         NaN   
15         NaN         NaN         NaN         NaN  ...         NaN   
16         NaN         NaN         NaN         NaN  ...         NaN   
17         NaN         NaN         NaN         NaN  ...         NaN   
18         NaN         NaN         NaN         NaN  ...         NaN   
19         NaN         NaN         NaN         NaN  ...         NaN   

    2020-12-23  2020-12-24  2020-12-25  2020-12-26  2020-12-27  2020-12-28  \
0        0.491         NaN         NaN         NaN       0.390       0.531   
1        0.493         NaN         NaN         NaN       0.368       0.515   
2        0.482         NaN         NaN         NaN       0.363       0.506   
3        0.483         NaN         NaN         NaN       0.375       0.505   
4        0.495         NaN         NaN         NaN       0.392       0.532   
5        0.519         NaN         NaN         NaN       0.406       0.565   
6        0.504         NaN         NaN         NaN       0.426       0.564   
7        0.513         NaN         NaN         NaN       0.407       0.538   
8        0.508         NaN       0.583         NaN       0.372       0.513   
9        0.491         NaN       0.549         NaN       0.358       0.500   
10       0.517         NaN         NaN         NaN       0.398       0.550   


In [8]:
Modis_data_2020.get_shape()

(1350, 368)

In [9]:
Modis_data_2021 = Modis_data_loader("MODIS_AOD/merged_data_2021_sorted_dates.csv")
Modis_data_2021.get_head()

latitude  longitude  2021-01-01  2021-01-02  2021-01-03  2021-01-04  \
0     28.414     77.171       0.586         NaN         NaN         NaN   
1     28.414     77.181       0.570         NaN         NaN         NaN   
2     28.414     77.192       0.561         NaN         NaN         NaN   
3     28.414     77.202       0.581         NaN         NaN         NaN   
4     28.414     77.213       0.595         NaN         NaN         NaN   
5     28.414     77.223         NaN         NaN         NaN         NaN   
6     28.424     77.171       0.623         NaN         NaN         NaN   
7     28.424     77.181       0.608         NaN         NaN         NaN   
8     28.424     77.192       0.583         NaN         NaN         NaN   
9     28.424     77.202       0.570         NaN         NaN         NaN   
10    28.424     77.213         NaN         NaN         NaN         NaN   
11    28.424     77.223         NaN         NaN         NaN         NaN   
12    28.424     77.233       0.597         NaN         NaN         NaN   
13    28.424     77.244       0.583         NaN         NaN         NaN   
14    28.434     77.161       0.601         NaN         NaN         NaN   
15    28.434     77.171       0.622         NaN         NaN         NaN   
16    28.434     77.181       0.648         NaN         NaN         NaN   
17    28.434     77.192       0.615         NaN         NaN         NaN   
18    28.434     77.202       0.594         NaN         NaN         NaN   
19    28.434     77.213       0.610         NaN         NaN         NaN   

    2021-01-05  2021-01-06  2021-01-07  2021-01-08  ...  2021-12-22  \
0          NaN         NaN         NaN         NaN  ...         NaN   
1          NaN         NaN         NaN         NaN  ...         NaN   
2          NaN         NaN         NaN         NaN  ...         NaN   
3          NaN         NaN         NaN         NaN  ...         NaN   
4          NaN         NaN         NaN         NaN  ...         NaN   
5          NaN         NaN         NaN         NaN  ...         NaN   
6          NaN         NaN         NaN         NaN  ...         NaN   
7          NaN         NaN         NaN         NaN  ...         NaN   
8          NaN         NaN         NaN         NaN  ...         NaN   
9          NaN         NaN         NaN         NaN  ...         NaN   
10         NaN         NaN         NaN         NaN  ...         NaN   
11         NaN         NaN         NaN         NaN  ...         NaN   
12         NaN         NaN         NaN         NaN  ...         NaN   
13         NaN         NaN         NaN         NaN  ...         NaN   
14         NaN         NaN         NaN         NaN  ...         NaN   
15         NaN         NaN         NaN         NaN  ...         NaN   
16         NaN         NaN         NaN         NaN  ...         NaN   
17         NaN         NaN         NaN         NaN  ...         NaN   
18         NaN         NaN         NaN         NaN  ...         NaN   
19         NaN         NaN         NaN         NaN  ...         NaN   

    2021-12-23  2021-12-24  2021-12-25  2021-12-26  2021-12-27  2021-12-28  \
0        1.931       1.516         NaN         NaN         NaN         NaN   
1        1.902       1.527         NaN         NaN         NaN         NaN   
2        1.869       1.577         NaN         NaN         NaN         NaN   
3        1.874       1.651         NaN         NaN         NaN         NaN   
4        1.892       1.589         NaN         NaN         NaN         NaN   
5        1.898       1.672         NaN         NaN         NaN         NaN   
6          NaN       1.573         NaN         NaN         NaN         NaN   
7        1.923       1.415         NaN         NaN         NaN         NaN   
8        1.890       1.510         NaN         NaN         NaN         NaN   
9        1.846       1.607         NaN         NaN         NaN         NaN   
10       1.859       1.669         NaN         NaN         NaN         NaN   


In [10]:
Modis_data_2021 = Modis_data_loader("MODIS_AOD/merged_data_2021_sorted_dates.csv")
Modis_data_2021.get_head()

latitude  longitude  2021-01-01  2021-01-02  2021-01-03  2021-01-04  \
0     28.414     77.171       0.586         NaN         NaN         NaN   
1     28.414     77.181       0.570         NaN         NaN         NaN   
2     28.414     77.192       0.561         NaN         NaN         NaN   
3     28.414     77.202       0.581         NaN         NaN         NaN   
4     28.414     77.213       0.595         NaN         NaN         NaN   
5     28.414     77.223         NaN         NaN         NaN         NaN   
6     28.424     77.171       0.623         NaN         NaN         NaN   
7     28.424     77.181       0.608         NaN         NaN         NaN   
8     28.424     77.192       0.583         NaN         NaN         NaN   
9     28.424     77.202       0.570         NaN         NaN         NaN   
10    28.424     77.213         NaN         NaN         NaN         NaN   
11    28.424     77.223         NaN         NaN         NaN         NaN   
12    28.424     77.233       0.597         NaN         NaN         NaN   
13    28.424     77.244       0.583         NaN         NaN         NaN   
14    28.434     77.161       0.601         NaN         NaN         NaN   
15    28.434     77.171       0.622         NaN         NaN         NaN   
16    28.434     77.181       0.648         NaN         NaN         NaN   
17    28.434     77.192       0.615         NaN         NaN         NaN   
18    28.434     77.202       0.594         NaN         NaN         NaN   
19    28.434     77.213       0.610         NaN         NaN         NaN   

    2021-01-05  2021-01-06  2021-01-07  2021-01-08  ...  2021-12-22  \
0          NaN         NaN         NaN         NaN  ...         NaN   
1          NaN         NaN         NaN         NaN  ...         NaN   
2          NaN         NaN         NaN         NaN  ...         NaN   
3          NaN         NaN         NaN         NaN  ...         NaN   
4          NaN         NaN         NaN         NaN  ...         NaN   
5          NaN         NaN         NaN         NaN  ...         NaN   
6          NaN         NaN         NaN         NaN  ...         NaN   
7          NaN         NaN         NaN         NaN  ...         NaN   
8          NaN         NaN         NaN         NaN  ...         NaN   
9          NaN         NaN         NaN         NaN  ...         NaN   
10         NaN         NaN         NaN         NaN  ...         NaN   
11         NaN         NaN         NaN         NaN  ...         NaN   
12         NaN         NaN         NaN         NaN  ...         NaN   
13         NaN         NaN         NaN         NaN  ...         NaN   
14         NaN         NaN         NaN         NaN  ...         NaN   
15         NaN         NaN         NaN         NaN  ...         NaN   
16         NaN         NaN         NaN         NaN  ...         NaN   
17         NaN         NaN         NaN         NaN  ...         NaN   
18         NaN         NaN         NaN         NaN  ...         NaN   
19         NaN         NaN         NaN         NaN  ...         NaN   

    2021-12-23  2021-12-24  2021-12-25  2021-12-26  2021-12-27  2021-12-28  \
0        1.931       1.516         NaN         NaN         NaN         NaN   
1        1.902       1.527         NaN         NaN         NaN         NaN   
2        1.869       1.577         NaN         NaN         NaN         NaN   
3        1.874       1.651         NaN         NaN         NaN         NaN   
4        1.892       1.589         NaN         NaN         NaN         NaN   
5        1.898       1.672         NaN         NaN         NaN         NaN   
6          NaN       1.573         NaN         NaN         NaN         NaN   
7        1.923       1.415         NaN         NaN         NaN         NaN   
8        1.890       1.510         NaN         NaN         NaN         NaN   
9        1.846       1.607         NaN         NaN         NaN         NaN   
10       1.859       1.669         NaN         NaN         NaN         NaN   


In [11]:
Modis_data_2021.get_shape()

(1350, 367)

In [12]:
Modis_data_2022 = Modis_data_loader("MODIS_AOD/merged_data_2022_sorted_dates.csv")
Modis_data_2022.get_head()

latitude  longitude  2021-12-31  2022-01-02  2022-01-03  2022-01-04  \
0     28.414     77.171         NaN       1.001         NaN         NaN   
1     28.414     77.181         NaN       0.898         NaN         NaN   
2     28.414     77.192         NaN       0.883         NaN         NaN   
3     28.414     77.202         NaN       0.867         NaN         NaN   
4     28.414     77.213         NaN       0.860         NaN         NaN   
5     28.414     77.223         NaN       0.868         NaN         NaN   
6     28.424     77.171         NaN       1.205         NaN         NaN   
7     28.424     77.181         NaN       1.024         NaN         NaN   
8     28.424     77.192         NaN       0.919         NaN         NaN   
9     28.424     77.202         NaN       0.895         NaN         NaN   
10    28.424     77.213         NaN       0.904         NaN         NaN   
11    28.424     77.223         NaN       0.922         NaN         NaN   
12    28.424     77.233         NaN       0.950         NaN         NaN   
13    28.424     77.244         NaN       0.965         NaN         NaN   
14    28.434     77.161         NaN       1.448         NaN         NaN   
15    28.434     77.171         NaN       1.352         NaN         NaN   
16    28.434     77.181         NaN       1.197         NaN         NaN   
17    28.434     77.192         NaN       0.943         NaN         NaN   
18    28.434     77.202         NaN       0.913         NaN         NaN   
19    28.434     77.213         NaN       0.909         NaN         NaN   

    2022-01-05  2022-01-06  2022-01-07  2022-01-08  ...  2022-12-22  \
0          NaN         NaN         NaN         NaN  ...         NaN   
1          NaN         NaN         NaN         NaN  ...         NaN   
2          NaN         NaN         NaN         NaN  ...         NaN   
3          NaN         NaN         NaN         NaN  ...         NaN   
4          NaN         NaN         NaN         NaN  ...         NaN   
5          NaN         NaN         NaN         NaN  ...         NaN   
6          NaN         NaN         NaN         NaN  ...         NaN   
7          NaN         NaN         NaN         NaN  ...         NaN   
8          NaN         NaN         NaN         NaN  ...         NaN   
9          NaN         NaN         NaN         NaN  ...         NaN   
10         NaN         NaN         NaN         NaN  ...         NaN   
11         NaN         NaN         NaN         NaN  ...         NaN   
12         NaN         NaN         NaN         NaN  ...         NaN   
13         NaN         NaN         NaN         NaN  ...         NaN   
14         NaN         NaN         NaN         NaN  ...         NaN   
15         NaN         NaN         NaN         NaN  ...         NaN   
16         NaN         NaN         NaN         NaN  ...         NaN   
17         NaN         NaN         NaN         NaN  ...         NaN   
18         NaN         NaN         NaN         NaN  ...         NaN   
19         NaN         NaN         NaN         NaN  ...         NaN   

    2022-12-23  2022-12-24  2022-12-25  2022-12-26  2022-12-27  2022-12-28  \
0        0.486         NaN         NaN         NaN       0.452         NaN   
1        0.513         NaN         NaN         NaN       0.477         NaN   
2        0.533         NaN         NaN         NaN       0.490         NaN   
3        0.545         NaN         NaN         NaN       0.505         NaN   
4        0.557         NaN         NaN         NaN       0.512         NaN   
5        0.592         NaN         NaN         NaN         NaN         NaN   
6        0.443         NaN         NaN         NaN       0.425         NaN   
7        0.486         NaN         NaN         NaN       0.461         NaN   
8        0.509         NaN         NaN         NaN       0.476         NaN   
9        0.534         NaN         NaN         NaN       0.489         NaN   
10         NaN         NaN         NaN         NaN       0.508         NaN   


In [13]:
Modis_data_2022.get_shape()

(1350, 367)

In [14]:
Modis_data_2023 = Modis_data_loader("MODIS_AOD/merged_data_2023_sorted_dates.csv")
Modis_data_2023.get_head()

latitude  longitude  2023-01-01  2023-01-02  2023-01-03  2023-01-04  \
0     28.414     77.171       0.317         NaN       0.567       0.339   
1     28.414     77.181       0.329         NaN       0.573       0.349   
2     28.414     77.192       0.334         NaN       0.571       0.363   
3     28.414     77.202       0.339         NaN       0.588       0.368   
4     28.414     77.213       0.345         NaN       0.605       0.373   
5     28.414     77.223       0.346         NaN       0.604       0.381   
6     28.424     77.171       0.316         NaN       0.571       0.332   
7     28.424     77.181       0.329         NaN       0.578       0.342   
8     28.424     77.192       0.326         NaN       0.579       0.355   
9     28.424     77.202       0.329         NaN       0.577       0.371   
10    28.424     77.213       0.341         NaN       0.619       0.373   
11    28.424     77.223         NaN         NaN       0.619       0.370   
12    28.424     77.233       0.323         NaN       0.604       0.370   
13    28.424     77.244       0.322         NaN       0.595       0.358   
14    28.434     77.161       0.275         NaN       0.541       0.352   
15    28.434     77.171       0.298         NaN       0.558       0.352   
16    28.434     77.181       0.315         NaN       0.575       0.356   
17    28.434     77.192       0.323         NaN       0.582       0.363   
18    28.434     77.202       0.327         NaN       0.583       0.381   
19    28.434     77.213       0.333         NaN       0.601       0.379   

    2023-01-05  2023-01-06  2023-01-07  2023-01-08  ...  2023-09-16  \
0        0.375       0.322         NaN         NaN  ...       1.128   
1        0.375       0.336         NaN         NaN  ...       1.141   
2        0.373       0.342         NaN         NaN  ...       1.142   
3        0.385       0.338         NaN         NaN  ...       1.175   
4        0.398       0.355         NaN         NaN  ...       1.211   
5        0.397       0.364         NaN         NaN  ...         NaN   
6        0.377       0.330         NaN         NaN  ...         NaN   
7        0.386       0.345         NaN         NaN  ...       1.153   
8        0.379       0.339         NaN         NaN  ...       1.146   
9        0.373       0.340         NaN         NaN  ...       1.132   
10       0.403       0.358         NaN         NaN  ...       1.209   
11       0.403       0.364         NaN         NaN  ...       1.247   
12       0.384       0.364         NaN         NaN  ...         NaN   
13       0.368       0.320         NaN         NaN  ...         NaN   
14       0.331       0.262         NaN         NaN  ...         NaN   
15       0.353       0.304         NaN         NaN  ...         NaN   
16       0.375       0.345         NaN         NaN  ...       1.169   
17       0.372       0.344         NaN         NaN  ...       1.120   
18       0.367       0.348         NaN         NaN  ...       1.113   
19       0.378       0.342         NaN         NaN  ...       1.146   

    2023-09-17  2023-09-18  2023-09-19  2023-09-20  2023-09-21  2023-09-22  \
0        0.707         NaN         NaN         NaN       0.438       0.465   
1        0.648         NaN         NaN         NaN       0.412       0.391   
2        0.638         NaN         NaN         NaN       0.384       0.394   
3        0.644         NaN         NaN         NaN       0.404       0.417   
4        0.702         NaN         NaN         NaN       0.431       0.472   
5        0.725         NaN         NaN         NaN       0.428       0.498   
6        0.793         NaN         NaN         NaN       0.451       0.533   
7        0.753         NaN         NaN         NaN       0.446       0.475   
8        0.693         NaN         NaN         NaN       0.410       0.411   
9        0.687         NaN         NaN         NaN       0.362       0.394   
10       0.761         NaN         NaN         NaN       0.422       0.474   


In [15]:
Modis_data_2023.get_shape()

(1350, 250)

### Latitude & Longitude data sets




In [16]:
Lat_Long_data = Modis_data_loader("Stations_Lat_Long/stn_extrafeat.csv")
# stations = Lat_Long_data.data[["Latitude", "Longitude"]]
# stations
Lat_Long_data.data

ID                                 Stn   Latitude  Longitude  Pop Density  \
0    0                              Alipur  28.815329  77.153010            1   
1    1                         Anand.Vihar  28.646835  77.316032            3   
2    2                         Ashok.Vihar  28.695381  77.181665            8   
3    3                           Aya.Nagar  28.470691  77.109936            1   
4    4                              Bawana  28.776200  77.051074            1   
5    5                     Burari.Crossing  28.725650  77.201157            2   
6    6                   CRRI.Mathura.Road  28.551201  77.273574            3   
7    7                       Chandni.Chowk  28.656756  77.227234            3   
8    8                                 DTU  28.750050  77.111261            1   
9    9      Dr..Karni.Singh.Shooting.Range  28.498571  77.264840            6   
10  10                              Dwarka  28.571027  77.071901            3   
11  11                    East.Arjun.Nagar  28.655602  77.285932            7   
12  12                    IGI.Airport.(T3)  28.562776  77.118005            1   
13  13               IHBAS..Dilshad.Garden  28.681174  77.302523            7   
14  14                                 ITO  28.631694  77.249439           10   
15  15                        Jahangirpuri  28.732820  77.170633            5   
16  16            Jawaharlal.Nehru.Stadium  28.580280  77.233829            3   
17  17                      Lodhi.RoadIITM  28.588333  77.221667            4   
18  18                       Lodhi.RoadIMD  28.591825  77.227307            4   
19  19  Major.Dhyan.Chand.National.Stadium  28.611281  77.237738            2   
20  20                         Mandir.Marg  28.636429  77.201067            8   
21  21                              Mundka  28.684678  77.076574            9   
22  22                         NSIT.Dwarka  28.609090  77.032541            2   
23  23                           Najafgarh  28.570173  76.933762            1   
24  24                              Narela  28.822836  77.101981            1   
25  25                         Nehru.Nagar  28.567890  77.250515            4   
26  26                    North.Campus..DU  28.657381  77.158545            6   
27  27                         Okhla.Phase  28.530785  77.271255            9   
28  28                          Patparganj  28.623748  77.287205            9   
29  29                        Punjabi.Bagh  28.563262  77.186937            7   
30  30                             PusaIMD  28.645000  77.167000            6   
31  31                            PusaDPCC  28.639645  77.146262            9   
32  32                           R.K.Puram  28.674045  77.131023            4   
33  33                              Rohini  28.732528  77.119920            4   
34  34                            Shadipur  28.651478  77.147311            6   
35  35                            Sirifort  28.550425  77.215938            7   
36  36                         Sonia.Vihar  28.710508  77.249485            4   
37  37                  Sri.Aurobindo.Marg  28.531346  77.190156            4   
38  38                         Vivek.Vihar  28.672342  77.315260            9   
39  39                            Wazirpur  28.699793  77.165453            6   

    Very dense  Mod dense  open Forest  
0     0.000250   0.016659     0.020568  
1     0.000000   0.017812     0.028906  
2     0.000250   0.016659     0.020568  
3     0.010440   0.054480     0.248360  
4     0.000250   0.016659     0.020568  
5     0.000000   0.050678     0.030847  
6     0.010440   0.054480     0.248360  
7     0.010440   0.054480     0.248360  
8     0.000250   0.016659     0.020568  
9     0.010440   0.054480     0.248360  
10    0.005629   0.030166     0.063492  
11    0.000000   0.017812     0.028906  
12    0.005629   0.030166     0.063492  
13    0.000000   0.018667     0.049667  
14    0.000000   0.089200     0.112800  
15    0.000250   0.016659 

In [17]:
stations = Lat_Long_data.data[["Latitude", "Longitude"]]
stations

Latitude  Longitude
0   28.815329  77.153010
1   28.646835  77.316032
2   28.695381  77.181665
3   28.470691  77.109936
4   28.776200  77.051074
5   28.725650  77.201157
6   28.551201  77.273574
7   28.656756  77.227234
8   28.750050  77.111261
9   28.498571  77.264840
10  28.571027  77.071901
11  28.655602  77.285932
12  28.562776  77.118005
13  28.681174  77.302523
14  28.631694  77.249439
15  28.732820  77.170633
16  28.580280  77.233829
17  28.588333  77.221667
18  28.591825  77.227307
19  28.611281  77.237738
20  28.636429  77.201067
21  28.684678  77.076574
22  28.609090  77.032541
23  28.570173  76.933762
24  28.822836  77.101981
25  28.567890  77.250515
26  28.657381  77.158545
27  28.530785  77.271255
28  28.623748  77.287205
29  28.563262  77.186937
30  28.645000  77.167000
31  28.639645  77.146262
32  28.674045  77.131023
33  28.732528  77.119920
34  28.651478  77.147311
35  28.550425  77.215938
36  28.710508  77.249485
37  28.531346  77.190156
38  28.672342  77.315260
39  28.699793  77.165453

## PM_2.5 data set


In [18]:
# Shape of the raw dataset.
PM_data_instance = PM_25_dataloader()
PM_data_instance.get_shape()


(40, 8760)

##### Shape of the data set is `(8760, 40, 16)` from here we have seen that there are 8760 hours in a year and 40 stations and 16 features in the data set but we have extracted the PM_25 data values for 40 stations and 8760 time stamps.

In [19]:
PM_data = PM_25_dataloader()
PM_data = PM_data.get_data()

# PM_data_with_Lat_long = combine_the_data_frames(stations, PM_data)
# PM_data_with_Lat_long.head()

PM_data.shape

(40, 8760)

### Training data


In [20]:
loader = Training_data_loader(Modis_data_2020.data, Modis_data_2021.data, Modis_data_2022.data, PM_data, stations)

### Extracting all the functional data from the Training data.


In [21]:
loader.daily_PM25_data_extraction()

2020-01-01  2020-01-02  2020-01-03  2020-01-04  2020-01-05  \
Column1   239.125000  217.708328  192.817535  225.343750  182.916672   
Column2   367.375031  361.687500  309.965179  235.875000  189.740250   
Column3   385.333344  376.291687  235.416641  282.729156  195.354156   
Column4   341.493317  185.562500  193.758331  166.028122  135.367294   
Column5   433.594391  318.809174  252.058319  267.624359  204.333328   
Column6   373.533752  339.496063  237.097839  255.008331  191.254944   
Column7   400.863556  348.464172  207.795410  203.410721  164.618530   
Column8   365.040588  328.674591  211.695633  221.190002  163.543793   
Column9   304.324371  251.275848  219.512726  257.288605  192.156845   
Column10  363.767944  379.616577  203.614609  203.927078  168.583344   
Column11  472.979187  455.218781  276.093750  214.885437  181.989578   
Column12  362.947205  327.655548  239.454636  222.427277  169.003586   
Column13  320.876678  291.574982  257.881653  179.122711  146.294373   
Column14  314.678131  264.956940  216.939575  189.439590  139.191666   
Column15  399.656250  400.065796  199.479172  224.250000  149.406250   
Column16  397.678741  386.481720  268.354156  299.656250  232.572922   
Column17  444.468689  350.010437  234.343750  209.958344  167.104172   
Column18  332.005615  271.294037  184.074768  171.575027  135.776184   
Column19  293.363525  236.003113  162.660416  151.656464  120.684586   
Column20  365.218781  344.244659  212.284821  204.913559  160.020828   
Column21  337.471100  301.806091  177.125000  209.239563  157.045380   
Column22  459.820129  337.718750  219.104172  265.708344  191.697922   
Column23  359.432495  385.085175  181.155609  183.310349  142.252487   
Column24  264.750000  276.812500  152.833344  143.229172  150.479172   
Column25  308.614594  260.645844  176.905472  191.625015  164.416672   
Column26  509.387695  438.197906  302.351837  281.333344  217.177078   
Column27  329.639221  275.014374  171.681030  203.107910  128.409180   
Column28  479.606415  368.875031  231.833328  219.625000  175.458328   
Column29  367.114594  335.031250  204.239594  209.708328  153.000000   
Column30  420.677063  364.729187  227.718750  243.645828  161.656265   
Column31  288.812500  279.437500  225.270844  238.739563  187.156250   
Column32  234.178543  208.848969  146.828323  164.331879  111.523117   
Column33  403.093750  283.687500  179.812485  138.781250  117.145836   
Column34  408.552094  393.072937  270.260406  285.416687  205.500000   
Column35  389.504120  385.112122  265.535400  261.536865  178.802704   
Column36  449.536926  334.386200  241.910004  204.434174  174.862000   
Column37  289.843750  214.322922  163.927078  204.906250  162.458328   
Column38  338.666656  215.501923  166.833328  161.864594  147.520828   
Column39  372.366882  332.208344  294.000000  265.430939  206.458328   
Column40  419.575562  397.399597  303.197937  290.291687  217.239594   

          2020-01-06  2020-01-07  2020-01-08  2020-01-09  2020-01-10  ...  \
Column1   169.093750   97.052086   87.039551   80.490089  112.875000  ...   
Column2   168.430511  142.376663  140.018799   95.112617  148.541672  ...   
Column3   169.354156  141.645828  132.346420  102.270836  142.187500  ...   
Column4   118.810417   94.127090   96.653534   62.955002   93.755005  ...   
Column5   207.416672  139.791672  115.250000  105.500000  147.166656  ...   
Column6   171.119156  135.287048  126.813019  108.765404  147.886810  ...   
Column7   146.270630  107.903534  119.322708   91.640419  138.901459  ...   
Column8   150.272781  124.257904  117.967934   89.779816  129.996765  ...   
Column9   170.519775  131.311249  110.489578   97.072083  140.195404  ...   
Column10  140.890274  118.388161  137.135422  101.468750  144.687500  ...   
Column11  182.302094  119.500000  121.352516  101.979172  144.425217  ...   
Column12  153.757294  124.951004  123.116302   88.692223  132.910309  ...   
Column13  140.661469   93.196251   89.037285   63.825211

In [22]:
loader.AOD_data_fusion()

latitude  longitude  2020-01-01  2020-01-02  2020-01-03  2020-01-04  \
0       28.414     77.171         NaN         NaN       1.428       1.092   
1       28.414     77.181         NaN         NaN       1.389       1.077   
2       28.414     77.192         NaN         NaN       1.325       1.085   
3       28.414     77.202         NaN         NaN       1.306       1.106   
4       28.414     77.213         NaN         NaN       1.312       1.157   
...        ...        ...         ...         ...         ...         ...   
1345    28.864     77.078         NaN         NaN         NaN       1.019   
1346    28.864     77.088         NaN         NaN       0.881       1.032   
1347    28.864     77.098         NaN         NaN       0.904       1.031   
1348    28.864     77.109         NaN         NaN       0.911       1.035   
1349    28.874     77.088         NaN         NaN       0.836       1.013   

      2020-01-05  2020-01-06  2020-01-07  2020-01-08  ...  2022-12-22  \
0            NaN         NaN         NaN         NaN  ...         NaN   
1            NaN         NaN         NaN         NaN  ...         NaN   
2            NaN         NaN         NaN         NaN  ...         NaN   
3            NaN         NaN         NaN         NaN  ...         NaN   
4            NaN         NaN         NaN         NaN  ...         NaN   
...          ...         ...         ...         ...  ...         ...   
1345         NaN         NaN         NaN         NaN  ...         NaN   
1346         NaN         NaN         NaN         NaN  ...         NaN   
1347         NaN         NaN         NaN         NaN  ...         NaN   
1348         NaN         NaN         NaN         NaN  ...         NaN   
1349         NaN         NaN         NaN         NaN  ...         NaN   

      2022-12-23  2022-12-24  2022-12-25  2022-12-26  2022-12-27  2022-12-28  \
0          0.486         NaN         NaN         NaN       0.452         NaN   
1          0.513         NaN         NaN         NaN       0.477         NaN   
2          0.533         NaN         NaN         NaN       0.490         NaN   
3          0.545         NaN         NaN         NaN       0.505         NaN   
4          0.557         NaN         NaN         NaN       0.512         NaN   
...          ...         ...         ...         ...         ...         ...   
1345         NaN         NaN         NaN         NaN       0.520         NaN   
1346         NaN         NaN         NaN         NaN       0.539         NaN   
1347         NaN         NaN         NaN         NaN       0.544         NaN   
1348         NaN         NaN         NaN         NaN       0.570         NaN   
1349         NaN         NaN         NaN         NaN         NaN         NaN   

      2022-12-29  2022-12-30  2022-12-31  
0            NaN         NaN         NaN  
1            NaN         NaN         NaN  
2            NaN         NaN         NaN  
3            NaN         NaN         NaN  
4            NaN         NaN         NaN  
...          ...         ...         ...  
1345         NaN         NaN         NaN  
1346         NaN         NaN         NaN  
1347         NaN         NaN         NaN  
1348         NaN         NaN         NaN  
1349         NaN         NaN         NaN  

[1350 rows x 1098 columns]

In [23]:
loader.PM25_nearest_neighbour_finder(loader.AOD_data_fusion())

latitude  longitude  2020-01-01  2020-01-02  2020-01-03  2020-01-04  \
0     28.815     77.150         NaN         NaN         NaN       1.228   
1     28.649     77.316         NaN         NaN       1.228       1.382   
2     28.698     77.181         NaN         NaN       1.214       1.053   
3     28.473     77.119         NaN         NaN       1.321       1.168   
4     28.776     77.046         NaN         NaN         NaN       1.196   
5     28.727     77.202         NaN         NaN         NaN       1.058   
6     28.551     77.275         NaN         NaN         NaN       1.194   
7     28.659     77.223         NaN         NaN       1.143       1.073   
8     28.747     77.109         NaN         NaN         NaN       1.222   
9     28.502     77.264         NaN         NaN         NaN       0.944   
10    28.571     77.067         NaN         NaN       1.170       1.054   
11    28.659     77.285         NaN         NaN       1.160       1.293   
12    28.561     77.119         NaN         NaN       1.232       0.986   
13    28.678     77.306         NaN         NaN       1.285       1.170   
14    28.629     77.254         NaN         NaN       1.221       1.360   
15    28.737     77.171         NaN         NaN         NaN       1.054   
16    28.580     77.233         NaN         NaN       1.310       1.212   
17    28.590     77.223         NaN         NaN       1.388       1.154   
18    28.590     77.223         NaN         NaN       1.388       1.154   
19    28.610     77.233         NaN         NaN       1.419       1.339   
20    28.639     77.202         NaN         NaN       1.189       1.189   
21    28.688     77.078         NaN         NaN         NaN       1.186   
22    28.610     77.036         NaN         NaN         NaN       1.203   
23    28.571     76.932         NaN         NaN         NaN       1.230   
24    28.825     77.098         NaN         NaN         NaN       1.180   
25    28.571     77.254         NaN         NaN       1.405       1.543   
26    28.659     77.161         NaN         NaN       1.164       1.098   
27    28.531     77.275         NaN         NaN       1.393       1.006   
28    28.619     77.285         NaN         NaN         NaN         NaN   
29    28.561     77.192         NaN         NaN       1.261       1.273   
30    28.649     77.171         NaN         NaN       1.100       1.152   
31    28.639     77.150         NaN         NaN       1.127       1.144   
32    28.678     77.129         NaN         NaN       1.260       1.222   
33    28.737     77.119         NaN         NaN         NaN       1.132   
34    28.649     77.150         NaN         NaN       1.174       1.106   
35    28.551     77.213         NaN         NaN       1.396       1.497   
36    28.707     77.254         NaN         NaN       1.125       1.019   
37    28.531     77.192         NaN         NaN       1.337       1.337   
38    28.668     77.316         NaN         NaN       1.235       1.234   
39    28.698     77.161         NaN         NaN         NaN       1.121   

    2020-01-05  2020-01-06  2020-01-07  2020-01-08  ...  2022-12-22  \
0          NaN         NaN         NaN         NaN  ...         NaN   
1          NaN         NaN         NaN         NaN  ...         NaN   
2          NaN         NaN         NaN         NaN  ...         NaN   
3          NaN         NaN         NaN         NaN  ...         NaN   
4          NaN         NaN         NaN         NaN  ...         NaN   
5          NaN         NaN         NaN         NaN  ...         NaN   
6          NaN         NaN         NaN         NaN  ...         NaN   
7          NaN         NaN         NaN         NaN  ...         NaN   
8          NaN         NaN         NaN         NaN  ...         NaN   
9          NaN         NaN         NaN         NaN  ...         NaN   
10         NaN         NaN         NaN         NaN  ...         NaN   
11         NaN         NaN         NaN         NaN  ...         NaN   
12   

In [24]:
loader.data_cleaning_and_missing_values_handeling(loader.PM25_nearest_neighbour_finder(loader.AOD_data_fusion()))

latitude  longitude  2020-01-01  2020-01-02  2020-01-03  2020-01-04  \
0     28.815     77.150         NaN         NaN         NaN       1.228   
1     28.649     77.316         NaN         NaN       1.228       1.382   
2     28.698     77.181         NaN         NaN       1.214       1.053   
3     28.473     77.119         NaN         NaN       1.321       1.168   
4     28.776     77.046         NaN         NaN         NaN       1.196   
5     28.727     77.202         NaN         NaN         NaN       1.058   
6     28.551     77.275         NaN         NaN         NaN       1.194   
7     28.659     77.223         NaN         NaN       1.143       1.073   
8     28.747     77.109         NaN         NaN         NaN       1.222   
9     28.502     77.264         NaN         NaN         NaN       0.944   
10    28.571     77.067         NaN         NaN       1.170       1.054   
11    28.659     77.285         NaN         NaN       1.160       1.293   
12    28.561     77.119         NaN         NaN       1.232       0.986   
13    28.678     77.306         NaN         NaN       1.285       1.170   
14    28.629     77.254         NaN         NaN       1.221       1.360   
15    28.737     77.171         NaN         NaN         NaN       1.054   
16    28.580     77.233         NaN         NaN       1.310       1.212   
17    28.590     77.223         NaN         NaN       1.388       1.154   
18    28.590     77.223         NaN         NaN       1.388       1.154   
19    28.610     77.233         NaN         NaN       1.419       1.339   
20    28.639     77.202         NaN         NaN       1.189       1.189   
21    28.688     77.078         NaN         NaN         NaN       1.186   
22    28.610     77.036         NaN         NaN         NaN       1.203   
23    28.571     76.932         NaN         NaN         NaN       1.230   
24    28.825     77.098         NaN         NaN         NaN       1.180   
25    28.571     77.254         NaN         NaN       1.405       1.543   
26    28.659     77.161         NaN         NaN       1.164       1.098   
27    28.531     77.275         NaN         NaN       1.393       1.006   
28    28.619     77.285         NaN         NaN         NaN         NaN   
29    28.561     77.192         NaN         NaN       1.261       1.273   
30    28.649     77.171         NaN         NaN       1.100       1.152   
31    28.639     77.150         NaN         NaN       1.127       1.144   
32    28.678     77.129         NaN         NaN       1.260       1.222   
33    28.737     77.119         NaN         NaN         NaN       1.132   
34    28.649     77.150         NaN         NaN       1.174       1.106   
35    28.551     77.213         NaN         NaN       1.396       1.497   
36    28.707     77.254         NaN         NaN       1.125       1.019   
37    28.531     77.192         NaN         NaN       1.337       1.337   
38    28.668     77.316         NaN         NaN       1.235       1.234   
39    28.698     77.161         NaN         NaN         NaN       1.121   

    2020-01-05  2020-01-06  2020-01-07  2020-01-08  ...  2022-12-22  \
0          NaN         NaN         NaN         NaN  ...         NaN   
1          NaN         NaN         NaN         NaN  ...         NaN   
2          NaN         NaN         NaN         NaN  ...         NaN   
3          NaN         NaN         NaN         NaN  ...         NaN   
4          NaN         NaN         NaN         NaN  ...         NaN   
5          NaN         NaN         NaN         NaN  ...         NaN   
6          NaN         NaN         NaN         NaN  ...         NaN   
7          NaN         NaN         NaN         NaN  ...         NaN   
8          NaN         NaN         NaN         NaN  ...         NaN   
9          NaN         NaN         NaN         NaN  ...         NaN   
10         NaN         NaN         NaN         NaN  ...         NaN   
11         NaN         NaN         NaN         NaN  ...         NaN   
12   

In [25]:
df = loader.get_final_training_data(loader.data_cleaning_and_missing_values_handeling(loader.PM25_nearest_neighbour_finder(loader.AOD_data_fusion())),loader.daily_PM25_data_extraction())
df.columns


Index(['latitude', 'longitude', 'date', 'AOD', 'PM2.5'], dtype='object')

In [26]:
final_training_data = loader.prepare_final_trainig_data()
final_training_data.shape

Extracting daily Pm2.5 data..
 daily_PM_data shape: (40, 1095)
Fusing AOD data from multiple years...
 AOD_final shape: (1350, 1098)
Finding nearest neighbours for PM2.5 stations...
nearest_neighbours shape: (40, 1098)
Cleaning data and handling missing values...
Cleaned nearest_neighbours shape: (40, 1097)
Preparing final training data...
final_training_data shape: (15667, 5)


(15667, 5)

In [27]:
final_training_data

latitude  longitude       date    AOD       PM2.5
0        28.815     77.150 2020-01-04  1.228  225.343750
1        28.815     77.150 2020-01-10  0.547  112.875000
2        28.815     77.150 2020-01-19  0.716  106.437500
3        28.815     77.150 2020-01-22  1.264  143.676773
4        28.815     77.150 2020-01-26  0.592  217.281235
...         ...        ...        ...    ...         ...
15662    28.698     77.161 2022-12-15  0.252   91.660454
15663    28.698     77.161 2022-12-17  0.288  181.042496
15664    28.698     77.161 2022-12-18  0.245  282.822418
15665    28.698     77.161 2022-12-20  0.736  210.184296
15666    28.698     77.161 2022-12-27  0.432  230.145844

[15667 rows x 5 columns]

### Final training data after using torch data loader
- This is the data from which we can train our 


In [28]:
from torch_data_loader import AirQualityDataset
from utils import config
import arrow
from datetime import datetime

{'dataset_num': 1, 'model': 'MLP', 'exp_name': 'aod_pm25_prediction', 'input_type_1': 1, 'input_type_2': 2, 'input_type_3': 3, 'target_type': ['PM2.5', 'AOD'], 'feature_sets': {1: ['latitude', 'longitude', 'AOD', 'PM2.5'], 2: ['latitude', 'longitude', 'AOD'], 3: ['latitude', 'longitude', 'PM2.5']}}


In [29]:
Instance = AirQualityDataset(final_training_data, config)

In [30]:
final_training_data = AirQualityDataset(final_training_data, config)

In [31]:
final_training_data.__getitem__(1)

(tensor([ 28.8150,  77.1500,   0.5470, 112.8750]),
 tensor([112.8750,   0.5470]))

In [32]:
final_training_data.__len__()

12307